### PR TITLE
Remove hardcoded URLs from store page and use changeState to display details

### PR DESCRIPTION
**Pull request by anthonydillon**

### CODE REVIEW

**Comment by anthonydillon** [comment]
## QA
- Click on the search box to enter the store
- Check the store looks the same as it did before
- Check some links and check they go to the correct place
- Click on a user or the docs link at the bottom and it should open in a new tab

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5501/
Test PASSed.

File: jujugui/static/gui/src/app/components/charmbrowser/charmbrowser.js [hunk 1]
```diff
@@ -25,9 +25,11 @@ YUI.add('charmbrowser-component', function() {
       acl: React.PropTypes.object.isRequired,
       addNotification: React.PropTypes.func.isRequired,
       apiUrl: React.PropTypes.string.isRequired,
+      apiVersion: React.PropTypes.string,
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/charmbrowser/charmbrowser.js]: Are these not required?

File: jujugui/static/gui/src/app/components/store/store.js [hunk 2]
```diff
@@ -22,7 +22,9 @@ YUI.add('store', function() {
 
   juju.components.Store = React.createClass({
     propTypes: {
+      apiVersion: React.PropTypes.string,
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/store/store.js]: ditto on the requirement

File: jujugui/static/gui/src/app/components/store/store.js [hunk 3]
```diff
@@ -74,6 +79,48 @@ YUI.add('store', function() {
...
     /**
+      Show the entity details when clicked.
+
+      @method _handleEntityClick
+      @param {String} id The entity id.
+      @param {Object} e The click event.
+    */
+    _handleEntityClick: function(e) {
+      e.stopPropagation();
+      var id = e.target.dataset.entity;
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/store/store.js]: unrelated, but interesting, jsdom doesn't support dataset :)

**Comment by hatched** [comment]
QA Issue:
When clicking on either the logo or the text in the following area the clicks do not work. You have to click on the whitespace. The same happens everywhere where this layout is used.

screenshot from 2016-07-27 10 28 31

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5504/
Test PASSed.

**Comment by anthonydillon** [comment]
@hatched Good spot, I meant to pick this one up once tests were passing but forgot. Fixed now.

File: jujugui/static/gui/src/app/components/charmbrowser/charmbrowser.js (hunk 1 above)
Comment (anthonydillon) [inline jujugui/static/gui/src/app/components/charmbrowser/charmbrowser.js]: done

File: jujugui/static/gui/src/app/components/store/store.js (hunk 2 above)
Comment (anthonydillon) [inline jujugui/static/gui/src/app/components/store/store.js]: done

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5515/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5516/
Test PASSed.

File: jujugui/static/gui/src/app/components/store/test-store.js [hunk 4]
```diff
@@ -30,702 +30,111 @@ describe('Store', function() {
...
-                        href="https://www.jujucharms.com/docs/stable/authors-charm-writing">
-                        Learn more about writing charms&nbsp;&rsaquo;
-                      </a></p>
-                  </div>
-              </div>
-          </div>
-      </div>);
-    /*eslint-enable max-len */
-    assert.deepEqual(output, expected);
+  it('can handle clicking on an search item', function() {
```
Comment (jaycee) [inline jujugui/static/gui/src/app/components/store/test-store.js]: s/on an search/on a search/

**Comment by jaycee** [comment]
Code looks :+1: to me, assuming @hatched blesses the QA.

**Comment by hatched** [comment]
Thanks for these updates @anthonydillon 

:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**